controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 		if err := r.deleteOnboardingSecret(); err != nil {
 			return reconcile.Result{}, nil
 		}
+
+		storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// isCephFsAvailable - get the default filesystem, isRbdAvailable - list blockpools CR with count 1, isNfsAvailable -
+		blockPoolList := &rookCephv1.CephBlockPoolList{}
+		if err := r.Client.List(r.ctx, blockPoolList, client.InNamespace(r.namespace), client.Limit(1)); err != nil {
+			return reconcile.Result{}, err
+		}
+		isRbdAvailable := len(blockPoolList.Items) > 0
```

Comment:
Checking based on blockpool list may not be a very good idea as the we have a blockpool created for NFS, one builtin for mgr